File: api/src/main/java/dev/obscuria/fragmentum/Platform.java
package dev.obscuria.fragmentum;

import java.nio.file.Path;

@SuppressWarnings("unused")
public interface Platform {

    Path getConfigDir();

    String getEnvironmentName();

    boolean isModLoaded(String modId);

    boolean isClient();

    boolean isDedicatedServer();

    boolean isDevelopmentEnvironment();
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/Platform.kt
@file:Suppress("unused", "RedundantNullableReturnType")

package dev.obscuria.fragmentum

interface Platform {

    fun getPlatformName(): String

    fun isModLoaded(modId: String): Boolean

    fun isClient(): Boolean

    fun isDedicatedServer(): Boolean = !isClient()

    fun isDevelopmentEnvironment(): Boolean
}